custom_components/frigate/services.yaml

```diff
@@ -30,3 +30,37 @@ favorite_event:
       default: true
       selector:
         boolean:
+
+ptz:
+  name: Control camera via PTZ
+  description: >
+    Pan / Tilt, Zoom, or move a camera to a preset
+  target:
+    entity:
+      integration: frigate
+      domain: camera
+      device_class: camera
+  fields:
+    service:
```

Comment:
It's a little confusing having a field called `service` in a service call which already has its own name. Thinking out loud perhaps it would be clearer to split the fields:

Tilt [optional]: up / down
Pan [optional]: left / right
Zoom [optional]: in / out
Preset [optional]: <name of preset>

... and we just act on whatever is specified. This is what ONVIF does: https://github.com/home-assistant/core/blob/dev/homeassistant/components/onvif/services.yaml

I think think this is probably a bunch more code complexity to "translate" between this protocol and what the backend is apparently expecting, so leave this with you to think about. Maybe there's a different naming that is a simpler fix, e.g. "action" (zoom/present/move) and "direction" (left/right/up/down/in/out/preset).

Thoughts?

Reply:
I really wanted to do that and I actually had the code that way, but I was worried about direction being non obvious for the preset usecase. If you don't think that is going to be an issue then I like that solution.

Reply:
Yes, I think that's the tradeoff. I think the ONVIF example is the maximally verbose and maximally clearest. What do you think about that one?

A middle option could be:

 - Action: Zoom / Preset / Move
 - Direction [optional]: Left / Right / Up / Down / In / Out
 - Present [optional]: <name>

But if you're going to do that, I think you might as well go the full ONVIF way, since "in" and "out" are not directions either.

Reply:
Generic alternative:
 - "Action": Zoom / Preset / Move
 - "Argument": <whatever>

Reply:
ONVIF way has a small advantage: You can do multiple movements in a single service call (e.g. move left and zoom in).

Reply:
I think the onvif way may make more sense in the future, but I don't really want to rework the whole backend implementation for the initial release, it is meant to be pretty simple out the gate. I also just realized that we need to add the `STOP` option here as well because frigate only implements the continuous move because many cameras I have been testing with don't support relative move.

Reply:
Sounds good with regards to leaving ONVIF verbosity for another day. I'm fine to leave it as service+action if that's your preference.

I assumed `left` / `right` were discrete steps to the left or right. Is that not the case, or is it starting a continuous motion that we need to stop?

Reply:
All of the actions besides preset start a move that then needs a consequent stop. For example in the frigate frontend clicking the button moves and then stop is sent once the click is released.

Reply:
OK. Stating the obvious this has the major disadvantage that HA processing time & network latency will impact the duration of the move and it's less automation friendly since users will need to work out the right delay between go and stop (and due to the latency, there really is no "right" delay). This will be less of a problem on the Frigate UI. I would imagine they will use presets to work around this though.

I could imagine another argument 'type' in the future that has `continuous` or `discrete`.

Reply:
Yeah, I definitely think for the initial implementation preset is the most useful for an automation. 

Another intermediary option could be to have a separate `duration` field, and the integration could handle sending the `stop` automatically for the move and zoom options

I agree there will be a lot of room to grow this feature

Reply:
> Another intermediary option could be to have a separate duration field, and the integration could handle sending the stop automatically for the move and zoom options

Yes! Although I'd slightly tweak that -- better to have the backend automatically handle the stop (vs the integration), since that would eliminate the HA->backend latency unpredictability.

Reply:
Yeah, there would just need to be a way to differentiate between the two in the payload. Although that work would likely be not as worth it with the additional of discrete movements.

I'll mark this as draft for now and think this through a bit more and see what I come up with.